e3sm_to_cmip/util.py

```diff
@@ -52,37 +50,345 @@ def print_message(message, status='error'):
+            datafile.computational_acknowledgement = str(
+                'The data were produced using resources of the National Energy Research Scientific Computing Center, a DOE Office of Science User Facility supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231.')
+            datafile.ncclimo_generation_command = str(
+                """ncclimo --var=${var} -7 --dfl_lvl=1 --no_cll_msr --no_frm_trm --no_stg_grd --yr_srt=1 --yr_end=500 --ypf=500 --map=map_ne30np4_to_cmip6_180x360_aave.20181001.nc """)
+            datafile.ncclimo_version = str('4.7.9')
+            pbar.update(idx)
+        finally:
+            datafile.close()
+    pbar.finish()
+# ------------------------------------------------------------------
+
+
+def find_atm_files(var, path):
+    """
+    Looks in the given path for all files that match that matche VAR_\d{6}_\d{6}.nc
+
+    Params:
+    -------
+        var (str): the name of the variable to look for
+        path (str): the path of the directory to look in
+    Returns:
+    --------
+        files (list(str)): A list of paths to the matching files
+    """
+    contents = os.listdir(path)
+    files = list()
+    pattern = '{}'.format(var) + r'\_\d{6}\_\d{6}.nc'
+    for item in contents:
+        if re.match(pattern=pattern, string=item):
+            files.append(item)
+    return files
+# ------------------------------------------------------------------
+
+
+def find_mpas_files(component, path):
+    """
+    Looks in the path given for MPAS monthly-averaged files
+
+    Params:
+    -------
+        component (str): Either the mpaso or mpassi component name
+        path (str): The path of the directory to search for files in
+    """
+    component = component.lower()
+    contents = os.listdir(path)
+
+    if component in ['mpaso', 'mpassi']:
+
+        pattern = '{}.hist.am.timeSeriesStatsMonthly.'.format(component) + \
+            r'\d{4}-\d{2}-\d{2}.nc'
+        return [os.path.join(path, x) for x in contents if re.match(
+            pattern=pattern, string=x)]
+
+    elif component == 'mpaso_namelist':
+
+        pattern = 'mpas-o_in'
+        for infile in contents:
+            if re.match(pattern, infile):
+                return [os.path.abspath(infile)]
+
+        raise IOError("Unable to find MPASO_namelist in the input directory")
+
+    elif component == 'mpassi_namelist':
+
+        pattern = 'mpas-cice_in'
+        for infile in contents:
+            if re.match(pattern, infile):
+                return [os.path.abspath(infile)]
+
+        raise IOError("Unable to find MPASSI_namelist in the input directory")
+
+    elif component == 'mpas_mesh':
+
+        # needs to be filled in by the driver
+        pattern = '{}.rst.'.format(component) + \
```

Comment:
```suggestion
        pattern = 'mpaso.rst.' + \
```
Both MPAS-O cand MPAS-SI can use an `mpaso` restart file as a mesh file.  The component is `mpas_mesh`, so the current version doesn't work.